### DISCUSSION TITLE
inline hook 的成功率

### DISCUSSION THREAD

**Discussion by CaichaoGitHub** [bytedance/android-inline-hook#73] [General]
inline hook 有线程安全的问题，我想知道是bytedance 的这套是如何保证hook的线程安全的
因为如果失败的话，系统会发段错误，复写的内存也脏了
<img width="675" alt="image" src="https://github.com/user-attachments/assets/ddeab10d-350c-472b-b7df-d17391bc745b">

**Comment by byte0100** [comment]
请问这个问题有新的进展吗？或者不会出现线程安全的问题吗？

**Comment by caikelun** [comment]
对于thumb指令是会有你说的多线程问题，可以看下手册的known issues部分。对于arm64和arm32指令，在shadowhook 2.0.0及以上的release版本中，不会有你说的多线程问题，因为shadowhook在被hook地址处永远是使用4字节相对跳转指令去替换原指令。
